Source/WebKit/Platform/IPC/HandleMessage.h

```diff
@@ -268,20 +361,39 @@ template<typename... MessageArgumentTypes>
 struct MethodSignatureValidationImpl<std::tuple<Connection&, MessageArgumentTypes...>, std::tuple<>>
 : MethodSignatureValidationImpl<std::tuple<MessageArgumentTypes...>, std::tuple<>> {
     static constexpr bool expectsConnectionArgument = true;
+    static constexpr bool expectsConnectionPointer = false;
+};
+
+template<typename... MessageArgumentTypes>
+struct MethodSignatureValidationImpl<std::tuple<Connection*, MessageArgumentTypes...>, std::tuple<>>
+: MethodSignatureValidationImpl<std::tuple<MessageArgumentTypes...>, std::tuple<>> {
+    static constexpr bool expectsConnectionArgument = true;
+    static constexpr bool expectsConnectionPointer = true;
 };
 
 template<typename... MessageArgumentTypes>
 struct MethodSignatureValidationImpl<std::tuple<MessageArgumentTypes...>, std::tuple<>> {
     static constexpr bool expectsConnectionArgument = false;
+    static constexpr bool expectsConnectionPointer = false;
     using MessageArguments = std::tuple<std::remove_cvref_t<MessageArgumentTypes>...>;
     using CompletionHandlerType = void;
+    static constexpr bool completionHandlerIsCelled = false;
```

Comment:
I plan to reword this as `completionHandlerIsReferenceCountableWrapped` or some such